components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/datasource/QueryExecutor.java

```diff
@@ -111,12 +110,13 @@ public boolean executeUpdateBusinessRuleQuery(String uuid, byte[] newBusinessRul
         PreparedStatement statement = null;
         try {
             conn = dataSource.getConnection();
-            conn.setAutoCommit(true);
+            conn.setAutoCommit(false);
```

Comment:
Please verify all the places and if you need to set it to false always, you can change the deployment.yaml data source configuration to isAutoCommit: false